linkerd/app/outbound/src/require_identity_on_endpoint.rs

```diff
@@ -155,57 +152,29 @@ where
             match self.peer_identity {
                 Conditional::Some(ref peer_identity) => {
                     if require_identity != *peer_identity {
-                        warn!(
```

Comment:
Should this still log a warning as well?

Reply:
No. We should _either_ warn _or_ return an error. The error is already log (and it now includes useful information, that it did not include before). I also do NOT want to log the required identity. This feels like an unnecessary info leak to me.